pkg/newlog/cmd/writelogFile.go

```diff
@@ -114,17 +124,28 @@ func writelogFile(logChan <-chan inputEntry, moveChan chan fileChanInfo) {
 
 			} else {
 				if entry.sendToRemote {
-					writelogEntry(&devStatsUpload, logline)
-
-					trigMoveToGzip(&devStatsUpload, "", moveChan, false)
+					if _, err := uploadSockWriter.Write([]byte(logline)); err != nil {
+						log.Errorf("writelogFile: uploadSockWriter write error: %v", err)
+					}
 				}
 
 				// write all log entries to the log file to keep
-				n := writelogEntry(&devStatsKeep, logline)
-				updateDevInputlogStats(entry.source, uint64(n))
+				n, err := keepSockWriter.Write([]byte(logline))
```

Comment:
Are we sure Vector is already listening at that moment? I don't see that we sync with the Vectore during the init. All the operations are running in goroutines, so I expect it may still not be running at the moment.
If Vector isn’t listening yet (has not started yet?), the first 1000 log lines fill the buffer, then Write will drop. Worth logging an explicit metric so operators notice.

Reply:
fixed it, now we wait for the vector sink sockets to appear  before we start sending logs to vector